### PR TITLE
fix: added contract symbol and position to selection context

### LINKED ISSUE #2905: Contract symbol selection is lost after clicking the trade or wallet button
Image

### PR DESCRIPTION
**Pull request by holzeis**
Introduces a selection context which stores the state of selection
 - contract symbol
 - position (long / short) per contract symbol

resolves #2905

### CODE REVIEW

File: taker-frontend/src/App.tsx [hunk 1]
```diff
@@ -62,24 +62,30 @@ const BTC_USD_SHORT_EVENT = "btcusd_short_offer";
...
+export interface Selection {
+    symbol: Symbol;
+    position: Map<Symbol, string>;
+}
+
+const defaultSelection: Selection = {
+    position: new Map().set(Symbol.btcusd, "long").set(Symbol.ethusd, "long"),
+    symbol: Symbol.btcusd,
+};
+export const SelectionContext = createContext(defaultSelection);
```
Comment (bonomat) [inline taker-frontend/src/App.tsx:74]: What do you think of giving this a better name? E.g. `SelectedSymbolContext`? Same goes for the interface `Selection` and the variable names.

**Review by bonomat — APPROVED** [review]
Nice, looks good.

File: taker-frontend/src/App.tsx (hunk 1 above)
Comment (holzeis) [reply taker-frontend/src/App.tsx:74]: It's not just the symbol but also the position 😉.

Comment (bonomat) [reply taker-frontend/src/App.tsx:74]: Still, `selection` does not say much :)

**Pull request merged**